xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -2467,7 +2466,6 @@ void CVideoPlayer::OnExit()
   SAFE_DELETE(m_pDemuxer);
   SAFE_DELETE(m_pSubtitleDemuxer);
   SAFE_DELETE(m_pCCDemuxer);
-  SAFE_DELETE(m_pInputStream);
```

Comment:
At ecery location vp relases inputsteam I want to make sure that is gets destructed. You could check usecount for being less or equal 1 before reset.